mix/qml/WebPreview.qml

```diff
@@ -184,8 +185,10 @@ Item {
 						content = fileIo.readFile(doc.path);
 				}
 
-				if (documentName === urlInput.text.replace(httpServer.url + "/", "")) {
-					//root page, inject deployment script
+				var accept = _request.headers["Accept"];
+				if (accept && accept.indexOf("text/html") >= 0 && !_request.headers["HTTP_X_REQUESTED_WITH"])
```

Comment:
This is do differentiate ajax requests from navigation requests. We only need to inject web3 object if the page request is coming from the browser, and not from javascript